server/jetstream_cluster.go

```diff
@@ -1383,6 +1386,13 @@ func (js *jetStream) applyStreamEntries(mset *Stream, ce *CommittedEntry, isReco
 				if err != nil {
 					panic(err.Error())
 				}
+				// Ignore if we are recovering and we have already processed.
+				if isRecovering && mset.State().LastSeq > sp.LastSeq {
+					// Make sure all messages from the purge are gone.
+					mset.store.Compact(sp.LastSeq)
```

Comment:
So I read that Compact() removes "up to but not including the seq parameter".
Is `LastSeq` in the purge request supposed to be included in the purge or is it the sequence in the stream that is supposed to be the new first after the purge? If the former, I wonder if it should not be `Compact(sp.LastSeq+1)` here then?

Reply:
Yes good point, will fix.